src/dmd/dcast.d

```diff
@@ -2740,11 +2740,15 @@ Expression scaleFactor(BinExp be, Scope* sc)
     {
         // Need to adjust operator by the stride
         // Replace (ptr + int) with (ptr + (int * stride))
-        Type t = Type.tptrdiff_t;
+        Type t = Type.tsize_t;
```

Comment:
DMD backend trips with this change. IndexExp implicit cast to this type instead of `ptrdiff_t`:

```
dmd: src/dmd/backend/cgcod.d:2533: Assertion `cast(int)(*e).Ecomsub <= cast(int)(*e).Ecount' failed.
```

Reply:
CC @WalterBright is this `assert` really necessary? It compiles and the test suite runs successfully without it, but I don't have knowledge in the backend to be sure about removing such code. See `test/runnable/mars1.d` for reference.